src/test/java/org/folio/edge/sip2/handlers/CheckoutHandlerTests.java

```diff
@@ -86,6 +87,61 @@ public void canExecuteASampleCheckoutUsingHandler(
         })));
   }
 
+  @Test
+  void cantExecuteASampleCheckoutUsingHandler(
+      @Mock CirculationRepository mockCirculationRepository,
+      Vertx vertx,
+      VertxTestContext testContext) {
+    final Clock clock = TestUtils.getUtcFixedClock();
+    final OffsetDateTime nbDueDate = OffsetDateTime.now();
+    final String institutionId = "diku";
+    final String patronIdentifier = "0192837465";
+    final String itemIdentifier = "1234567890";
+    final Checkout checkout = Checkout.builder()
+        .scRenewalPolicy(FALSE)
+        .noBlock(FALSE)
+        .transactionDate(OffsetDateTime.now())
+        .nbDueDate(nbDueDate)
+        .institutionId(institutionId)
+        .patronIdentifier(patronIdentifier)
+        .itemIdentifier(itemIdentifier)
+        .terminalPassword("1234")
+        .itemProperties("Some property of this item")
+        .patronPassword("4321")
+        .feeAcknowledged(FALSE)
+        .cancel(FALSE)
+        .build();
+
+    when(mockCirculationRepository.performCheckoutCommand(any(), any()))
+        .thenReturn(Future.failedFuture(new ClientException("Incorrect Username")));
+
+    final CheckoutHandler handler = new CheckoutHandler(mockCirculationRepository,
+        FreemarkerRepository.getInstance().getFreemarkerTemplate(CHECKOUT_RESPONSE));
+
+    final SessionData sessionData = TestUtils.getMockedSessionData();
+    sessionData.setPatronPasswordVerificationRequired(TRUE);
+    sessionData.setErrorResponseMessage(CheckoutResponse.builder()
+        .ok(FALSE)
+        .renewalOk(FALSE)
+        .magneticMedia(null)
+        .desensitize(TRUE)
+        .transactionDate(OffsetDateTime.now(clock))
+        .institutionId(institutionId)
+        .patronIdentifier(patronIdentifier)
+        .itemIdentifier(itemIdentifier)
+        .titleIdentifier("Some Book")
+        .dueDate(OffsetDateTime.now(clock).plusDays(30))
+        .build());
+
+    handler.execute(checkout, sessionData).onComplete(
+        testContext.failing(sipMessage -> testContext.verify(() -> {
+
+          assertEquals("Incorrect Username", sipMessage.getMessage());
+
+          testContext.completeNow();
+        })));
+  }
```

Comment:
I'm not completely sure how this test differs from the one below. I mean I can see the additional properties, and I can see the method name is different. Would some comments in the body of the test make it more clear what it is demonstrating?

Reply:
This test code covers the code of lines present in CheckOutHandler class when the ClientExcecption is thrown with the error message "Incorrect Username."
The incorrectUsernameFailure test case is for EndPatronSessionHandler.